File: frontend-v2/src/services/competitorIntel.ts
import api from './api'

export interface CompetitorPredictInput {
  name: string
  industry: string
  region: string
  size: string
  historical_discount?: number
}

export interface CompetitorIntelPredictPayload {
  project_id?: string
  budget?: number
  project_type?: string
  competitors: CompetitorPredictInput[]
}

export interface DiscountDistribution {
  p10: number
  p50: number
  p90: number
  distribution_type: string
  alpha: number
  beta_param: number
}

export interface EvidenceItem {
  source: string
  reliability: string
  date: string
  detail: string
}

export interface AccompliceAlert {
  risk_level: 'high' | 'medium' | 'low'
  reasons: string[]
  related_companies: string[]
  anomaly_score: number
}

export interface CompetitorPrediction {
  name: string
  point_estimate: number
  distribution: DiscountDistribution
  confidence: number
  evidence: EvidenceItem[]
  accomplice_probability: number
  accomplice_alert: AccompliceAlert | null
  market_position: string
}

export interface CompetitorIntelPredictResponse {
  predictions: CompetitorPrediction[]
  accomplice_groups: string[][]
}

export async function predictCompetitorIntel(payload: CompetitorIntelPredictPayload): Promise<CompetitorIntelPredictResponse> {
  const res = await api.post<CompetitorIntelPredictResponse>('/competitor-intel/predict', payload)
  return res.data
}

File: backend-v2/app/schemas/competitor_intel.py
from pydantic import BaseModel, Field
from typing import Optional


class CompetitorPredictInput(BaseModel):
    name: str
    industry: str = ""
    region: str = ""
    size: str = ""
    historical_discount: Optional[float] = None


class CompetitorIntelPredictRequest(BaseModel):
    project_id: Optional[str] = None
    budget: float = Field(ge=0, default=0)
    project_type: str = ""
    competitors: list[CompetitorPredictInput] = []


class DiscountDistribution(BaseModel):
    p10: float
    p50: float
    p90: float
    distribution_type: str = "beta"
    alpha: float = 0.0
    beta_param: float = 0.0


class EvidenceItem(BaseModel):
    source: str
    reliability: str = "C"
    date: str = ""
    detail: str = ""


class AccompliceAlert(BaseModel):
    risk_level: str = Field(default="low", description="high / medium / low")
    reasons: list[str] = []
    related_companies: list[str] = []
    anomaly_score: float = Field(ge=0, le=1, default=0.0)


class CompetitorPrediction(BaseModel):
    name: str
    point_estimate: float
    distribution: DiscountDistribution
    confidence: float = Field(ge=0, le=1, default=0.5)
    evidence: list[EvidenceItem] = []
    accomplice_probability: float = Field(ge=0, le=1, default=0.0)
    accomplice_alert: Optional[AccompliceAlert] = None
    market_position: str = "balanced"


class CompetitorIntelPredictResponse(BaseModel):
    predictions: list[CompetitorPrediction] = []
    accomplice_groups: list[list[str]] = []
